packages/mobile/src/paymentRequest/IncomingPaymentRequestListItem.tsx

```diff
@@ -91,24 +91,36 @@ export default function IncomingPaymentRequestListItem({ id, amount, comment, re
     }
   }
 
+  useFocusEffect(
+    React.useCallback(() => {
+      const removeButtonFocusListener = navigation.addListener('focus', () => {
+        setPayButtonPressed(false)
+        setAddressesFetched(false)
+      })
+
+      return removeButtonFocusListener()
```

Comment:
I think we need to return the function without calling it here, since it will be called by React when the screen loses focus.